synapse/events/spamcheck.py

```diff
@@ -26,24 +26,25 @@
 
 class SpamChecker(object):
     def __init__(self, hs: "synapse.server.HomeServer"):
-        self.spam_checker = None
-
-        module = None
-        config = None
-        try:
-            module, config = hs.config.spam_checker
-        except Exception:
-            pass
-
-        if module is not None:
-            # Older spam checkers don't accept the `api` argument, so we
-            # try and detect support.
-            spam_args = inspect.getfullargspec(module)
-            if "api" in spam_args.args:
-                api = SpamCheckerApi(hs)
-                self.spam_checker = module(config=config, api=api)
-            else:
-                self.spam_checker = module(config=config)
+        self.spam_checkers = []  # type: List[Any]
+
+        for spam_checker in hs.config.spam_checkers:
+            module = None
+            config = None
+            try:
+                module, config = spam_checker
+            except Exception:
+                pass
```

Comment:
This confuses me -- why are we catching all exceptions here?

Frankly i think I'd change this code to `for module, config in hs.config.spam_checkers` and go from there. We set `hs.config.spam_checkers` from other code we control so each entry should always be a valid two tuple.

Reply:
(Also, I know that this code is essentially just copied from before, but we should improve it while we're here!)